core/src/test/java/feast/core/service/JobCoordinatorServiceTest.java

```diff
@@ -84,221 +90,480 @@ public void shouldDoNothingIfNoStoresFound() throws InvalidProtocolBufferExcepti
         new Job(
+            "id2",
             "",
-            "extId2",
             Runner.DATAFLOW,
-            feast.core.model.Source.fromProto(source2),
-            feast.core.model.Store.fromProto(store),
+            source2,
+            feast.core.model.Store.fromProto(storeSpec),
             Arrays.asList(featureSet2),
             JobStatus.PENDING);
 
     Job expected2 =
         new Job(
-            "name2",
-            "extId2",
+            "id2",
+            "extid2",
             Runner.DATAFLOW,
-            feast.core.model.Source.fromProto(source2),
-            feast.core.model.Store.fromProto(store),
+            source2,
+            feast.core.model.Store.fromProto(storeSpec),
             Arrays.asList(featureSet2),
             JobStatus.RUNNING);
     ArgumentCaptor<List<Job>> jobArgCaptor = ArgumentCaptor.forClass(List.class);
 
     when(featureSetRepository.findAllByNameLikeAndProject_NameLikeOrderByNameAsc("%", "project1"))
         .thenReturn(Lists.newArrayList(featureSet1, featureSet2));
-
+    when(sourceRepository.findFirstByTypeAndConfigOrderByIdAsc(
+            source1.getType(), source1.getConfig()))
+        .thenReturn(source1);
+    when(sourceRepository.findFirstByTypeAndConfigOrderByIdAsc(
+            source2.getType(), source2.getConfig()))
+        .thenReturn(source2);
     when(specService.listStores(any()))
-        .thenReturn(ListStoresResponse.newBuilder().addStore(store).build());
+        .thenReturn(ListStoresResponse.newBuilder().addStore(storeSpec).build());
 
     when(jobManager.startJob(argThat(new JobMatcher(expectedInput1)))).thenReturn(expected1);
     when(jobManager.startJob(argThat(new JobMatcher(expectedInput2)))).thenReturn(expected2);
     when(jobManager.getRunnerType()).thenReturn(Runner.DATAFLOW);
 
+    when(jobRepository.findByStatus(JobStatus.RUNNING)).thenReturn(List.of());
+    when(jobRepository.findFirstBySourceIdAndStoreNameAndStatusNotInOrderByLastUpdatedDesc(
+            source1.getId(), storeSpec.getName(), JobStatus.getTerminalStates()))
+        .thenReturn(Optional.empty());
+    when(jobRepository.findFirstBySourceIdAndStoreNameAndStatusNotInOrderByLastUpdatedDesc(
+            source2.getId(), storeSpec.getName(), JobStatus.getTerminalStates()))
+        .thenReturn(Optional.empty());
+
     JobCoordinatorService jcs =
         new JobCoordinatorService(
-            jobRepository, featureSetRepository, specService, jobManager, feastProperties);
+            jobRepository,
+            featureSetRepository,
+            sourceRepository,
+            specService,
+            jobManager,
+            feastProperties);
     jcs.Poll();
 
     verify(jobRepository, times(1)).saveAll(jobArgCaptor.capture());
     List<Job> actual = jobArgCaptor.getValue();
 
-    assertThat(actual.get(0), equalTo(expected1));
-    assertThat(actual.get(1), equalTo(expected2));
+    assertThat(actual, containsInAnyOrder(expected1, expected2));
+  }
+
+  @Test
```

Comment:
please write test with two stores subscripted to different sources

Reply:
Added test.